offer-manager-domain/src/main/groovy/life/qbic/business/products/create/CreateProduct.groovy

```diff
@@ -28,15 +31,36 @@ class CreateProduct implements CreateProductInput {
     @Override
     void create(Product product) {
         try {
-            //todo create new productId!!! the id is not provided from the controller. It must be generated in the use case itself
-            dataSource.store(product)
-            output.created(product)
+            Product newProduct = buildNewProduct(product)
```

Comment:
Suggestion: create a copy constructor?
```suggestion
            Product newProduct = buildNewProduct(product)
```

Reply:
Everything that happens here is the id creation, so I removed this function :)